topostats/io.py

```diff
@@ -449,6 +452,49 @@ def convert_basename_to_relative_paths(df: pd.DataFrame):
     return df
 
 
+class Scale:
```

Comment:
I've been thinking about this and am wondering if making this a class is perhaps over-complicating things?

Scaling is a simple multiplication really, its just a case of selecting the correct scaling factor based on the units used in the scans and what is desired.

We can have sensible defaults defined in the configuration which also allows users to extend the possible scalings and could also be setup to allow the selection of the desired scaling using a simple function to select the correct options from the dictionary.

I realise this is a fairly poorly formed idea and lacks code but what do you think @iobataya ?

Reply:
Hi @iobataya 

This all sounds eminently reasonable and bulks out my above comments with more solid ideas, thank you.

I appreciate this is somewhat divergent to your original PR and am grateful of you taking the time to work through these things.

A couple of things that might be useful.

## Merge conflicts

As we've had a couple of merges recently there are now merge conflicts, I'll resolve these on GitHub but you will need to update your local branch with `git pull` to pull those changes down.

## Development tools

You may want to install some of the development tools we have listed as optional dependencies and make sure `pre-commit` is installed and running.

Dev tools can be installed if you do the following in your virtual environment (assuming you are using one, if not this might be a useful resource [Conda environments for effective and reproducible research](https://rse.shef.ac.uk/conda-environments-for-effective-and-reproducible-research/))...

```bash
cd ~/path/to/TopoStats
pip install -e .[dev]
```

This will pull in `pytest-cov` and tell you what the coverage change is although don't worry too much we have GitHub working with CodeCov to report these things.

However, we are increasingly trying to adhere to [PEP8 code style](https://pep8.org) and [NumpyDocstyle](https://numpydoc.readthedocs.io/en/latest/format.html) for docstrings and have [pre-commit](https://pre-commit.com) hooks in place which when enabled will check changes are compliant prior to allowing permits (we also use [pre-commit.ci](https://pre-commit.ci) to check pull requests but its a faster feedback loop to enable `pre-commit` locally).

Once you've installed the `dev` requirements you can in your VirtualEnv install the `pre-commit` hooks with `pre-commit install` and then when you make commits the checks will run and tell you if there are any problems (some will be automatically corrected if possible). I've a blog post I've written on pre-commit that you may find useful [pre-commit: Protecting your future self](https://rse.shef.ac.uk/blog/pre-commit) if you want to find out more.

Please do let me know if you've any problems, more than happy to help and could even have a chat over video if you think that would be useful.

@ns-rse

Reply:
Hello @ns-rse,

I appreciate your info about `git pull@ and `pip install -e .`. Especially, `pip install -e .` saved tons of time. And `pytest-cov` works fine to show coverage of codes. The test_io.py covers 84%.

I'm happy that I've reached the topostats in my branch can load gwy of any sizes of image (square and non-square) with arbitary pixel/lines aspect ratio. Now this makes me to proceed my project. I'll continue to improve the codes, and format codes to the expected style.

Reply:
Brilliant, that is great to hear @iobataya :+1: 

You'll notice below that there are a couple of merge conflicts that have arisen. This arises when changes have been made to files after you forked the repository and Git doesn't know which should be kept.

They need manually resolving, probably the easiest is to do that here on GitHub using the links/buttons below. You then choose which bit of code should be retained and a commit will be made to your branch here on GitHub which you then `git pull` down to your local copy.

Let me know if you've any problems or would like help with this.

Reply:
@ns-rse  Sorry for my silence for a while. I prepared for domestic conference commencing tomorrow. My program using TopoStats is public now (https://github.com/iobataya/line_detection). Once the conference and other events are over and I take a break, I will start organizing the codes here.

Reply:
Thanks for the update @iobataya I hope the conference went well.